internal/store/store.go

```diff
@@ -89,10 +87,7 @@ func (s *lockingStore) UpsertGateway(ctx context.Context, gateway Gateway, updat
 	}
 
 	// Attempt to bind all Route(s) to the Gateway in case any are newly-able to bind
-	_, boundRoutes, err := s.bindAll(ctx, []Gateway{gateway}, routes)
-	if err != nil {
-		return err
-	}
+	_, boundRoutes := s.bindAll(ctx, []Gateway{gateway}, routes)
```

Comment:
`bindAll` and the binding functions that make it up never actually returned error values. Since this is an internal function, we can always add back if we need to later, but this made things easier to reason about for now